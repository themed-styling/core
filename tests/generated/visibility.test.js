import 'jest-styled-components'

import { visibility } from '../../src/index.js'

import { testAllOn } from '../utilities.js'

testAllOn(visibility, 'visibility')
