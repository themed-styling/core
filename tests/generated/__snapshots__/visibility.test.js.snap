// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`visibility renders 0 1`] = `
Array [
  .c0 {
  visibility: 0;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: 0;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: 0;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: 0;
}

<div
    className="c0"
  />,
]
`;

exports[`visibility renders function 1`] = `
Array [
  .c0 {
  visibility: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: has;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: has;
}

<div
    className="c0"
  />,
]
`;

exports[`visibility renders invalid theme path 1`] = `
Array [
  .c0 {
  visibility: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: invalid.path.1;
}

<div
    className="c0"
  />,
]
`;

exports[`visibility renders negative number 1`] = `
Array [
  .c0 {
  visibility: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: -42365;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: -42365;
}

<div
    className="c0"
  />,
]
`;

exports[`visibility renders object [breakpoints] 1`] = `
Array [
  @media screen and (min-width:November) {
  .c0 {
    visibility: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    visibility: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    visibility: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    visibility: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {
  .c0 {
    visibility: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    visibility: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    visibility: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    visibility: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    visibility: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    visibility: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    visibility: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    visibility: November;
  }
}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    visibility: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    visibility: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    visibility: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    visibility: November;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`visibility renders object [breakpoints] 2`] = `
Array [
  @media May {
  .c0 {
    visibility: July;
  }
}

@media April {
  .c0 {
    visibility: August;
  }
}

@media January {
  .c0 {
    visibility: September;
  }
}

@media April {
  .c0 {
    visibility: November;
  }
}

@media June {
  .c0 {
    visibility: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {
  .c0 {
    visibility: July;
  }
}

@media April {
  .c0 {
    visibility: August;
  }
}

@media January {
  .c0 {
    visibility: September;
  }
}

@media April {
  .c0 {
    visibility: November;
  }
}

@media June {
  .c0 {
    visibility: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    visibility: January;
  }
}

@media April {
  .c0 {
    visibility: December;
  }
}

@media January {
  .c0 {
    visibility: April;
  }
}

@media April {
  .c0 {
    visibility: May;
  }
}

@media June {
  .c0 {
    visibility: June;
  }
}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    visibility: January;
  }
}

@media April {
  .c0 {
    visibility: December;
  }
}

@media January {
  .c0 {
    visibility: April;
  }
}

@media April {
  .c0 {
    visibility: May;
  }
}

@media June {
  .c0 {
    visibility: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`visibility renders postitive number 1`] = `
Array [
  .c0 {
  visibility: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: 42365;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: 42365;
}

<div
    className="c0"
  />,
]
`;

exports[`visibility renders string 1`] = `
Array [
  .c0 {
  visibility: July;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: July;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: June;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: June;
}

<div
    className="c0"
  />,
]
`;

exports[`visibility renders valid theme path 1`] = `
Array [
  .c0 {
  visibility: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  visibility: DarkSlateBlue;
}

<div
    className="c0"
  />,
]
`;
